ext/simdjson/src/bindings.cpp

```diff
@@ -0,0 +1,238 @@
+/*
+  +----------------------------------------------------------------------+
+  | simdjson_php                                                         |
+  +----------------------------------------------------------------------+
+  | This source file is subject to version 2.0 of the Apache license,    |
+  | that is bundled with this package in the file LICENSE, and is        |
+  | available through the world-wide-web at the following url:           |
+  | http://www.apache.org/licenses/LICENSE-2.0.html                      |
+  +----------------------------------------------------------------------+
+  | Author: Jinxi Wang  <1054636713@qq.com>                              |
+  | Author: Máté Kocsis <kocsismate@woohoolabs.com>                      |
+  +----------------------------------------------------------------------+
+*/
+
+#include "php.h"
+#include "php_simdjson.h"
+
+#include "simdjson.h"
+#include "bindings.h"
+
+extern "C" {
+	#include <ext/json/php_json.h>
+	#include <Zend/zend_exceptions.h>
+}
+
+// see https://github.com/simdjson/simdjson/blob/master/doc/performance.md#reusing-the-parser-for-maximum-efficiency
+simdjson::dom::parser parser;
+
+WARN_UNUSED
+simdjson::error_code
+build_parsed_json_cust(
+	simdjson::dom::element &doc, const char *buf, size_t len, bool realloc_if_needed, u_short depth = simdjson::DEFAULT_MAX_DEPTH
+) {
+    auto error = parser.allocate(len, depth);
+
+    if (error) {
+        return error;
+    }
+
+    error = parser.parse(buf, len, realloc_if_needed).get(doc);
+    if (error) {
+        return error;
+    }
+
+    return simdjson::SUCCESS;
+}
+
+static zval create_array(simdjson::dom::element element)
+{
+    zval v;
+
+    switch (element.type()) {
+        //ASCII sort
+        case simdjson::dom::element_type::STRING :
+            ZVAL_STRING(&v, std::string_view(element).data());
+            break;
+        case simdjson::dom::element_type::INT64 : ZVAL_LONG(&v, int64_t(element));
+            break;
+        case simdjson::dom::element_type::UINT64 : ZVAL_LONG(&v, uint64_t(element));
```

Comment:
For this to work as expected, you'd

1. need to add extra code for 32-bit builds of PHP and convert out of range values to doubles (`#if PHP_INT_SIZE == 4`)
2. Need to check if the value `> ZEND_LONG_MAX` and convert to double for out of range UINT64/INT64